src/bp/core/services/telemetry/configs.ts

```diff
@@ -0,0 +1,170 @@
+    @inject(TYPES.JobService) jobService: JobService,
+    @inject(TYPES.TelemetryRepository) telemetryRepo: TelemetryRepository,
+    @inject(TYPES.BotService) private botService: BotService,
+    @inject(TYPES.ModuleLoader) private moduleLoader: ModuleLoader,
+    @inject(TYPES.ConfigProvider) private config: ConfigProvider
+  ) {
+    super(ghostService, database, licenseService, jobService, telemetryRepo)
+    this.url = process.TELEMETRY_URL
+    this.lock = 'botpress:telemetry-configs'
+    this.interval = ms('7d')
+  }
+
+  protected async getStats(): Promise<TelemetryEvent> {
+    return {
+      ...buildSchema(await this.getServerStats(), 'server'),
+      event_type: 'configs',
+      event_data: {
+        schema: '1.0.0',
+        botConfigs: await this.getBotsConfigs(),
+        modulesConfigs: await this.getModulesConfigs(),
+        botpressConfig: await this.getBotpressConfigs()
+      }
+    }
+  }
+
+  private obfuscateSecrets(config, defaultConfig): any {
+    return _.reduce(
+      config,
+      (res, value, key) => {
+        if (SECRET_KEYS.find(x => key.toLowerCase().includes(x))) {
+          res[key] = _.isEqual(config[key], defaultConfig[key]) ? DEFAULT : REDACTED
+        } else if (!_.isArray(value) && _.isObject(value)) {
+          res[key] = this.obfuscateSecrets(value, defaultConfig[key])
+        } else {
+          res[key] = value
+        }
+        return res
+      },
+      {}
+    )
+  }
+
+  private async fetchSchema(schemaName: string): Promise<any> {
+    try {
+      return this.ghostService.root().readFileAsObject('/', schemaName)
+    } catch (error) {
+      return {}
+    }
+  }
+
+  private async getBotpressConfigs(): Promise<BotpressConfig> {
+    const botpressConfig = await this.config.getBotpressConfig()
+    const defaultConfig = defaultJsonBuilder(await this.fetchSchema('botpress.config.schema.json'))
+
+    return this.obfuscateSecrets(botpressConfig, defaultConfig)
+  }
+
+  private async getModulesConfigs(): Promise<ModuleConfigEvent[]> {
+    const bots = await this.botService.getBotsIds()
+    const modules = _.intersection(BUILTIN_MODULES, Object.keys(process.LOADED_MODULES))
+
+    return (await Promise.map(modules, this.getConfigsByModule(bots))).reduce((acc, cur) => [...acc, ...cur])
+  }
+
+  private getConfigsByModule(bots: string[]): (module: string) => Promise<ModuleConfigEvent[]> {
+    return async module => {
+      const defaultValue = await this.moduleLoader.configReader.loadFromDefaultValues(module)
+      return Promise.map(bots, this.getModuleConfigPerBot(module, defaultValue))
+    }
+  }
+
+  private getModuleConfigPerBot(module: string, defaultValue: any): (botId: any) => Promise<ModuleConfigEvent> {
+    return async botId => {
+      const runtimeValue = await this.moduleLoader.configReader.getForBot(module, botId)
+      return { botId, module, configs: this.obfuscateSecrets(runtimeValue, defaultValue) }
+    }
+  }
```

Comment:
Can you avoid adding empty module configurations? It will just take space for nothing.

![image](https://user-images.githubusercontent.com/42552874/89306319-b2d6eb80-d63d-11ea-9d20-9468a902c5c7.png)

Reply:
fixed: 216f7b05d3c7c5fa8efef056528fa987d9c28660